app/code/Magento/Ui/DataProvider/AbstractDataProvider.php

```diff
@@ -162,6 +162,8 @@ public function getFieldMetaInfo($fieldSetName, $fieldName)
 
     /**
      * @inheritdoc
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
```

Comment:
why do we need this suppress annotation?

Reply:
There is return type declared for this. method in interface
```
    /**
     * Add field filter to collection
     *
     * @param \Magento\Framework\Api\Filter $filter
     * @return mixed
     */
    public function addFilter(\Magento\Framework\Api\Filter $filter);
```
So we either should suppress it or update interface. 
Detected by static test
```
2) Magento\Test\Php\LiveCodeTest::testPhpStan
PHPStan detected violation(s):
 ------ ------------------------------------------------------------------ 
  Line   app/code/Magento/Ui/DataProvider/AbstractDataProvider.php         
 ------ ------------------------------------------------------------------ 
  168    Method Magento\Ui\DataProvider\AbstractDataProvider::addFilter()  
         should return mixed but return statement is missing.              
 ------ ------------------------------------------------------------------ 
```
Do you think we should process thought approval process and update interface ? I believe update to annotation in interface will cause SVC failure (It looks like we don't need return type in this method)

Reply:
we only need MissingReturnTypeHint , I've removed this extra annotation